src/app/api/services/dot-events.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { DotEvent } from '../../shared/models/event/dot-event';
+
+/**
+ * Provide a Global service to Subscribe to custom events and notify subscribers when those events occur.
+ * @DotEventsService
+ */
+@Injectable()
+export class DotEventsService {
```

Comment:
I am not aware about the intention here and I am not an expert on TS, neither angular; however it seems your Service only has the ability to subscribe one subscriber. If it is a singleton over the app I would give the ability to add more than one listener.
However and as I said, I am not an angular guy so not sure if my comment is out of context and I am wrong

Reply:
The `BehaviorSubject ` overwrite the current value with a new one each time the `notify` method is called, and all the subscribers are listening to the `BehaviorSubject ` in a specific event type. In general has the ability to subscribe and react as many events as required.